packages/astro/src/runtime/client/visible.ts

```diff
@@ -11,8 +11,11 @@ const visibleDirective: ClientDirective = (load, options, el) => {
 		await hydrate();
 	};
 
-	const ioOptions = {
-		rootMargin: typeof options.value === 'string' ? options.value : undefined,
+	const rawOptions =
+		typeof options.value === 'object' ? (options.value as ClientVisibleOptions) : undefined;
```

Comment:
Final nit: this _technically_ adds support for all `IntersectionObserverInit` values, but our types are narrowed to only support `Pick<IntersectionObserverInit, 'rootMargin'>`. Is that desired?

Reply:
It doesn't, see the code below